warehouse/legacy/api/pypi.py

```diff
@@ -145,3 +154,35 @@ def display(request):
             request.route_path("packaging.release", name=name, version=version)
         )
     return HTTPMovedPermanently(request.route_path("packaging.project", name=name))
+
+
+@view_config(route_name="legacy.api.pypi.token.new")
+def create_token(request):
+    macaroon_service = request.find_service(IMacaroonService, context=None)
+    try:
+        macaroon_service.verify(
+            raw_macaroon=request.master_key,
+            context="",
+            principals="",
+            permission={"version": 2, "permissions": "user"},
+        )
+    except InvalidMacaroon:
+        raise HTTPUnauthorized()
+    user = macaroon_service.find_userid(
+        request.master_key
```

Comment:
I see that this `master_key` attribute gets mocked in the tests, but I can't find where it actually appears in the (real) request object. Does it get set anywhere, or is this WIP?